src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java

```diff
@@ -275,11 +297,15 @@ else if (msg.what == KILL_WORKER) {
 
                     ///////////////////////////
 
-                    if (queueDepth >= mConfig.getBulkUploadLimit()) {
+                    if (queueDepth >= mConfig.getBulkUploadLimit() && SystemClock.uptimeMillis() >= mRetryAfter) {
```

Comment:
There is an interesting and strange edge case- what if we have only a few records, but they're very, very large? I'd like to try and push here under the following circumstances
```
if ( (queueDepth >= mConfig.getBulkUploadLimit() || we are dropping events due to lack of space) &&
    SystemClock.elapsedRealtime() >= mRetryAfter) {
...
}
```